src/vs/workbench/api/common/extHostTerminalService.ts

```diff
@@ -639,11 +651,27 @@ export abstract class BaseExtHostTerminalService extends Disposable implements I
 		if (!profile || !('options' in profile)) {
 			throw new Error(`No terminal profile options provided for id "${id}"`);
 		}
+
+		const internalOptions: ITerminalInternalOptions = this._resolveParentTerminal(options);
```

Comment:
If we convert to/from ExtHostTerminalIdentifier whenever we go between mainThread and extHost, can we avoid this?

Reply:
no because this is the very way we are able to pass between them. perhaps I'm not understanding